File: 2155-find-missing-observations/2155-find-missing-observations.js
/**
 * @param {number[]} rolls
 * @param {number} mean
 * @param {number} n
 * @return {number[]}
 */
var missingRolls = function(rolls, mean, n) {
    const sumRolls = rolls.reduce((acc, roll) => acc + roll, 0);
    const m = rolls.length;
    const missingSum = mean * (n + m) - sumRolls;

    if (missingSum < n || missingSum > n * 6) return [];
    
    const val = Math.floor(missingSum / n);
    let rem = missingSum % n;
    const ans = Array(n).fill(val);
    
    for (let i = 0; i < rem; i++) {
        ans[i]++;
    }
    
    return ans;
};

File: 2155-find-missing-observations/2155-find-missing-observations.cpp
class Solution {
public:
    vector<int> missingRolls(vector<int>& rolls, int mean, int n) {
        int m = rolls.size();
        int len = n + m;
        int sum = accumulate(rolls.begin(), rolls.end(), 0);
        vector<int> ans;
        int remainingSum = mean * len - sum;
        
        int currVal = remainingSum/n;
        int rem = remainingSum % n;
        if (remainingSum < n || remainingSum > 6 * n) return ans;

        ans.resize(n);
        for(int i = 0; i<n; i++) {
            if(rem > 0) {
                ans[i] = currVal + 1;
                rem--;
            }
            else ans[i] = currVal;
        }

        return ans;
    }
};